### PR TITLE
Implement performance metrics collector workflow

### PR DESCRIPTION
**Pull request by torredil**
**What is this PR about? / Why do we need it?**

This PR implements a workflow to collect driver deployment time, container resource usage, and volume lifecycle performance during CI runs with the goal of preventing silent regressions over time. At a high level, the metrics collector script:

- Collects driver deployment time, resource usage metrics, and clusterloader2 metrics.
- Configures bucket policy to allow the associated Lambda function to access the bucket.
- Sets a lifecycle policy on the S3 bucket to automatically expire metrics data after 3 days.
- Configures bucket notification to trigger the Lambda function on object creation events.
- Uploads the collected metrics to the S3 bucket.

**What testing is done?** 

```
I0612 14:54:08.101590   27309 phase_latency.go:146] PodStartupLatency: perc50: 11.72959236s, perc90: 14.944266027s, perc99: 14.944266027s; threshold 5m0s
I0612 14:54:08.101598   27309 phase_latency.go:146] PodStartupLatency: perc50: 4s, perc90: 4s, perc99: 4s
I0612 14:54:18.302610   27309 clusterloader.go:248] --------------------------------------------------------------------------------
I0612 14:54:18.302627   27309 clusterloader.go:249] Test Finished
I0612 14:54:18.302634   27309 clusterloader.go:250]   Test: ../aws-ebs-csi-driver/hack/e2e/metrics/cl2-config.yaml
I0612 14:54:18.302640   27309 clusterloader.go:251]   Status: Success
I0612 14:54:18.302645   27309 clusterloader.go:255] --------------------------------------------------------------------------------

2023-06-12 14:54:33 [INFO] - Clusterloader2 tests completed successfully
2023-06-12 14:54:33 [INFO] - Checking if S3 bucket aws-ebs-csi-driver-metrics exists
2023-06-12 14:54:34 [INFO] - Configuring bucket policy to allow Lambda access
2023-06-12 14:54:36 [INFO] - Setting lifecycle policy on S3 bucket aws-ebs-csi-driver-metrics
2023-06-12 14:54:37 [INFO] - Configuring bucket notification to trigger Lambda function
2023-06-12 14:54:39 [INFO] - Uploading metrics to S3 bucket aws-ebs-csi-driver-metrics
```

### CODE REVIEW

File: hack/e2e/metrics/override.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# This file contains override configurations for the cluster loader test.
+# It specifies various settings for provisioning volumes, waiting for PVs to be bound or deleted,
+# pod and volume configurations, and the expected provisioner and storage class.
+
+PROVISION_VOLUME: true
```
Comment (hanyuel) [inline hack/e2e/metrics/override.yaml:5]: nit: do we not have these defaulting to true? in [hack/e2e/metrics/cl2-config.yaml](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1624/files#diff-bcab731453d2b4b230a0181cad15f0dc0b6474f48e47f8e6b5dc7162772714e0)

File: hack/e2e/run.sh [hunk 2]
```diff
@@ -82,6 +83,8 @@ HELM_CT_TEST=${HELM_CT_TEST:-"false"}
 CHART_TESTING_VERSION=${CHART_TESTING_VERSION:-3.7.1}
 CLEAN=${CLEAN:-"true"}
 
+METRICS=${METRICS:-"false"}
```
Comment (ConnorJC3) [inline hack/e2e/run.sh:86]: Can we consider naming this something like `COLLECT_METRICS`, to make it more apparent what it does?

File: hack/e2e/run.sh [hunk 3]
```diff
@@ -267,6 +270,16 @@ fi
 
 OVERALL_TEST_PASSED="${TEST_PASSED}"
 
+if [[ "${METRICS}" == true ]]; then
+  metrics_collector "$KUBECONFIG" \
+    "$AWS_ACCOUNT_ID" \
+    "$AWS_REGION" \
+    "$NODE_OS_DISTRO" \
+    "$secondUsed" \
```
Comment (ConnorJC3) [inline hack/e2e/run.sh:278]: I know this isn't your fault, but `$secondUsed` is such a bad name, can we change it to something like `$deployTimeSeconds` or whatever?

**Review by ConnorJC3 — COMMENTED** [review]
Mostly lgtm some nits

**Review by ConnorJC3 — APPROVED** [review]
lgtm

**Pull request merged**